packages/shared/components/inputs/AssetDropdown.svelte

```diff
@@ -1,16 +1,28 @@
 <script lang="typescript">
     import { NetworkProtocol } from '@core/network'
+    import { localize } from '@core/i18n'
     import { visibleSelectedAccountAssets } from '@core/wallet'
-    import { AssetTile, Icon, Text, AssetIcon } from 'shared/components'
+    import { AssetIcon, AssetTile, Icon, InputContainer, Text, TextInput } from 'shared/components'
     import { FontWeight } from 'shared/components/Text.svelte'
     import { clickOutside } from 'shared/lib/actions'
 
     export let asset = $visibleSelectedAccountAssets?.baseCoin
+    export let isFocused = false
+    export let isDropdownOpen = false
+    export let inputElement: HTMLInputElement
 
-    let isDropdownOpen = false
+    let searchQuery = ''
     let icon: string
 
     $: hasMultipleAssets = $visibleSelectedAccountAssets?.nativeTokens.length >= 1
+    $: searchQuery = searchQuery.toLocaleLowerCase()
+    $: dropdownAssetList = [$visibleSelectedAccountAssets?.baseCoin]
+        .concat($visibleSelectedAccountAssets?.nativeTokens)
+        .filter((asset) => {
+            const assetName = asset?.metadata?.name?.toLocaleLowerCase()
+            const assetUnit = asset?.metadata?.unit?.toLocaleLowerCase()
```

Comment:
```suggestion
            const assetName = asset.metadata.name.toLocaleLowerCase()
            const assetUnit = asset.metadata.unit.toLocaleLowerCase()
```
i think we don't need these `?`

Reply:
Would we need it after metadata like 'asset.metadata?.unit.toLocaleLowerCase()` since metadata is an optional property in the `IPersistedAsset` interface?

Reply:
not 100% sure, because i think that only valid assets (and those have to have metadata) are included in this list. 
But to be safe, just let the metadata?.